08.ls_object/dir_info.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+require 'etc'
+require_relative './content_producer'
+
+class DirInfo
+  include ContentProducer
+  def initialize(dir_path, options)
+    @target_contents = fetch_contents(dir_path, options)
+    @original_dir_path = dir_path
+    @dir_path = Dir.exist?(dir_path) ? "#{File.expand_path(dir_path)}/" : ''
+  end
+
+  def produce_dir_info
+    if @target_contents.nil?
+      puts format('ls.rb: %s: No such file or directory', @original_dir_path)
+      return
+    end
+
+    target_info, block_size = produce_target_info
+    each_info_length = fetch_each_info_length(target_info)
+    formatted_target_info = format_target_info(target_info, each_info_length)
+    output_info_lists(formatted_target_info, block_size)
+  end
+
+  private
+
+  def produce_target_info
+    half_year_ago = Time.new - 24 * 60 * 60 * 180
+    block_size = 0
+    target_info =
+      @target_contents.map do |t_content|
+        target_content = @dir_path + t_content
+        target_info = File.stat(target_content)
+        block_size += target_info.blocks
+        {
+          target_type: File.ftype(target_content)[0],
+          target_permission: transform_permission(target_info.mode.to_s(8)[-3..]),
+          nlink: target_info.nlink.to_s,
+          target_user_name: Etc.getpwuid(target_info.uid).name,
+          target_group_name: Etc.getgrgid(target_info.gid).name,
+          target_size: target_info.size.to_s,
+          created_time: target_info.mtime >= half_year_ago ? target_info.mtime.strftime('%_m %_d %H:%M') : target_info.mtime.strftime('%_m %_d  %_Y'),
+          basename: @target_contents.count == 1 && FileTest.file?(@dir_path) ? @original_target : File.basename(target_content)
```

Comment:
ブロックを伴わないので size でいいと思います。

```suggestion
          basename: @target_contents.size == 1 && FileTest.file?(@dir_path) ? @original_target : File.basename(target_content)
```